### PR TITLE
#3430 Warning for changes in another page editor instance in a different browser tab

### LINKED ISSUE #3430: Warn when editing a blueprint/extension that's open in a different Page Editor tab
Context
- When working with multi-page blueprints, it's tempting to open up the Page Editor on multiple tabs
- The Page Editor instances aren't kept in sync, so it's easy to overwrite/lose work
- Keeping Page Editor instances in sync would be a big undertaking, with tricky behavior around how to reflect/merge changes across instances
- We want to warn the user before the make changes/put the tabs into a divergent state. (It would be frustrating to warn them only on save)

Implementation Sketch
- Add a Redux slice for mirroring extension/blueprint active+dirty state
- Keep that slice in sync across tabs -- there's likely an existing library? Or can use the extension's localStorage? (Does our existing redux-persistor already support the use case?

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Closes #3430 

## Reviewer Tips

- Look at 'sessionChangesSlice`
- Then look at `sessionChangesListenerMiddleware`, where the slice is updated
- Finally, look at `<StaleSessionPane/>`, and `<EditorLayout/>` where slice is read and the pane rendered

## Demo

- https://www.loom.com/share/d8903170d21a489ab3de647cd182c7af

## Future Work

- We might consider keeping the page editor states in sync in the future, but that requires some work around making the UI state logic more robust

## Checklist

- [x] Add tests
- [ ] Run Storybook and manually confirm that all stories are working
- [x] Designate a primary reviewer - @BALEHOK

### CODE REVIEW

**Comment by corinnemayans** [comment]
Suggested updated warning text:

There were changes made in a different instance of the Page Editor. Reload this Page Editor to sync the changes.
[Reload]

1. This would also apply if a team member has made an edit, right?
2. After clicking "Reload", it is a bit disjointed with warning going away and showing the in-page-editor message "Loading Page Editor" (which is kind of small?) followed by the "Loading Data..." modal. I don't think we need to resolve this right now, but wanted to call it out as a bit funky.

File: src/pageEditor/sessionChanges/sessionChangesSelectors.test.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { uuidv4 } from "@/types/helpers";
+import { selectIsStaleSession } from "@/pageEditor/sessionChanges/sessionChangesSelectors";
+
+describe("selectIsStaleSession", () => {
+  test("empty changes state", () => {
+    const state: SessionRootState & SessionChangesRootState = {
+      session: {
+        sessionId: uuidv4(),
+        sessionStart: Date.now(),
+      },
```
[the remark this answers is not in the document]
Comment (BALEHOK) [reply src/pageEditor/sessionChanges/sessionChangesSelectors.test.ts:29]: Same for `sessionChanges`

**Review by BALEHOK — APPROVED** [review]
See Corinne's comment about the copy.

**Comment by BLoe** [comment]
@corinnemayans 

>After clicking "Reload", it is a bit disjointed with warning going away and showing the in-page-editor message "Loading Page Editor" (which is kind of small?) followed by the "Loading Data..." modal. I don't think we need to resolve this right now, but wanted to call it out as a bit funky.

Yeah this is definitely a little messy, but those loading indicators are out of scope of this task, they already exist in the page editor, they weren't added or changed for this. Definitely something we can look into cleaning up separately though.

**Pull request merged**